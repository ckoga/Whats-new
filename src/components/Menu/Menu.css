.nav-bar {
  display: flex;
  flex-direction: column;
}

.nav-button {
  background: #141527;
  border: none;
  color: white;
  height: 40px;
  font-size: 1em;
  margin: 5px 10px;
  border-radius: 0 20px 20px 0;
  width: 90%;
}

.nav-button:hover {
  background-color: #FEAD00;
  cursor: pointer;
}