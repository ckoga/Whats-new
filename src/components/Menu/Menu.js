import React, { Component } from 'react';
import './Menu.css'

// MENU COMPONENT CODE GOES HERE
class Menu extends Component {
  constructor() {
    super()
  }

  selectCategory = (event) => {
    event.preventDefault();
    this.props.pickCategory(event.target.dataset.category)
  }

  render() {
    return (
    <nav className="nav-bar">
      <button
        className="nav-button"
        data-category="local"
        onClick={event => this.selectCategory(event)}>Local
      </button>
      <button
        className="nav-button"
        data-category="technology"
        onClick={event => this.selectCategory(event)}>Technology
      </button>
      <button
        className="nav-button"
        data-category="entertainment"
        onClick={event => this.selectCategory(event)}>Entertainment
      </button>
      <button
        className="nav-button"
        data-category="science"
        onClick={event => this.selectCategory(event)}>Science
      </button>
      <button
        className="nav-button"
        data-category="health"
        onClick={event => this.selectCategory(event)}>Health
      </button>
    </nav>
    )
  }
}


export default Menu;