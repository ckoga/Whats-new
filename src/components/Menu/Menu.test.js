import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  it('should render the nav to the DOM', () => {
    const wrapper = shallow(<Menu />)

    expect(wrapper).toMatchSnapshot();
  });

})