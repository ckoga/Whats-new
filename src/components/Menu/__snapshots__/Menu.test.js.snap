// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render the nav to the DOM 1`] = `
<nav
  className="nav-bar"
>
  <button
    className="nav-button"
    data-category="local"
    onClick={[Function]}
  >
    Local
  </button>
  <button
    className="nav-button"
    data-category="technology"
    onClick={[Function]}
  >
    Technology
  </button>
  <button
    className="nav-button"
    data-category="entertainment"
    onClick={[Function]}
  >
    Entertainment
  </button>
  <button
    className="nav-button"
    data-category="science"
    onClick={[Function]}
  >
    Science
  </button>
  <button
    className="nav-button"
    data-category="health"
    onClick={[Function]}
  >
    Health
  </button>
</nav>
`;
